lib/wallet/send.dart

```diff
@@ -53,30 +53,28 @@ class _SendState extends State<Send> {
                       children: [
                         Row(
                           mainAxisAlignment: MainAxisAlignment.end,
-                          children: [
-                            ElevatedButton(
-                                onPressed: () async {
-                                  FLog.info(text: "Sending invoice" + invoice);
-                                  String? error;
-                                  try {
-                                    await api.sendLightningPayment(invoice: invoice);
-                                  } on FfiException catch (e) {
-                                    error = e.message;
-                                  }
-
-                                  ScaffoldMessenger.of(context).showSnackBar(SnackBar(
-                                    content: Text(error == null
-                                        ? "Paid lightning invoice $invoice"
-                                        : "Failed to send payment: $error"),
-                                  ));
-                                  context.go('/');
-                                },
-                                child: const Text('Send'))
-                          ],
+                          children: [AsyncButton(onPressedFunction: send, label: "Pay Invoice")],
                         ),
                       ],
                     ),
                   ),
                 ]))));
   }
+
+  void send() {
+    FLog.info(text: "Paying invoice $invoice");
+    api.sendLightningPayment(invoice: invoice).then((value) async {
```

Comment:
❓ Shouldn't we add an await here somewhere?

Reply:
I am happy to stick to the `await` and `try ... catch` syntax - I don't know why we introduced different syntax using the `.then` combinator.

either way we should stick to one pattern I think. At the moment it is confusing, because in some places we have `try ...  await ...  catch` and in others we have `.then ... .error`.

Reply:
According to @holzeis, if you use `.then` instead of `await`, you don't wait for the future to resolve.

Reply:
Hm... that's not how I would understand [the docs](https://api.flutter.dev/flutter/dart-async/Future/then.html). My understanding is that `then` is only called when the future resolves, which implies for me that it is awaited.

I find the `try ... catch` syntax more readable anyway, so I'll change this.

Reply:
Okay, this is what Richie was referring to: https://stackoverflow.com/questions/37551843/invoking-async-function-without-await-in-dart-like-starting-a-thread.

As I understand it, if you omit the `await` the future runs, but you continue and don't care about the future.

That is why you'd want to attach combinators in that case, to handle what happens without waiting for it to happen!

---

This is obviously different to how futures work in Rust, where you have to `await` if you want the future to be polled.

Reply:
Interesting caveat from that thread:

> This isn't like a thread though. As mentioned processing is enqueued to the event queue which means it won't be executed until the current task and all previously enqueued tasks are completed.
>
> If you want real parallel execution you need to utilize isolates.

Reply:
Ah, OK! I think I was confused by the syntax, but I can add the `await` regardless of the syntax to await the *returned* `Future`! Thanks for the link, that helped. 

We can actually keep the syntax, but add the `await`, from what I understand from the link you sent it will indeed change the behavior. 
I wonder what is the most idiomatic way to await and lifecycle management of dart futures in general. I could also return a future from the helper functions I created and then `await` when calling them on the `onPressed` function of the button - would we expect an `await` there?